### DISCUSSION TITLE
Freitextfeld bei CustomTools

### DISCUSSION THREAD

**Discussion by Andreas-Lampert** [e-netze/webgis-community#25] [Ideas]
Hallo!

Wäre es möglich bei CustomTools als Input Type zusätzlich zum type: 'select' auch ein Freitextfeld zu konfigurieren?
Z.B. könnte beim Höhenprofil zusätzlich auch der Profilname übergeben werden. 

`command: 'https://voibos.rechenraum.com/voibos/voibos?name=profilservice&beschriftung={profilname}&ueberhoehung={ueberhoehung}&hintergrund={hintergrund}&stuetzpunktabstand={stuetzpunktabstand}&polygonzug={wkt}&crs=31259&output=Profil,Lage',`

Mit dem untenstehenden Eintrag (type: 'input') ist es möglich ein Textfeld zu konfigurieren, es passt allerdings von der Formatierung noch nicht bzw. wird bei der Eingabe des Buchstabens "S" das Snapping geöffnet. 

`{ type: 'label', label: 'Profilname' },
         {
             id: 'profilname',
             label: 'Profilname',
             type: 'input',
         },`

Nice to have für das Input-Feld:
- Konfiguration eines Default values
- Angabe eines pattern zur Überprüfung des Eingabewerts (z.B. dürfte beim Überhöhungsfaktor kein Buchstabe drin stehen)
- Einschränkung des Eingabewerts (ein Überhöhungsfaktor von 2000 macht keinen Sinn)
- Maximale Anzahl der Zeichen
- Angabe ob Pflichtfeld/Optional

Lg, Andi

**Comment by gis-eni** [comment]
Eingabefelder sind bereits möglich, ich hab die Online Doku schon dementsprechend erweitert:

https://docs.webgiscloud.com/cloud/webgis_apps/mapviewer/customjs/customtools.html#benutzerdefinierte-werkzeuge-mit-eingabefeldern

"Nice to have" werde ich mir beizeiten einmal anschauen und eventuell ergänzen. Default Value könnte mit ``{ type: "input-text", value: "default value" }`` bereits funktionieren.

**Reply by Andreas-Lampert** [reply]
Vielen Dank! Funktioniert super.
